CustomLinkedList/src/main/java/CustomLinkedList.java

```diff
@@ -0,0 +1,129 @@
+public class CustomLinkedList<V> {
+    private int size = 0;
+    private Node<V> first;
+    private Node<V> last;
+
+    public int size() {
+        return size;
+    }
+
+    public boolean addFirst(V value) {
+        Node<V> newNode = new Node<>(value);
+        newNode.setNext(first);
+        if (first != null) {
+            first.setPrev(newNode);
+        } else {
+            last = newNode;
+        }
+        first = newNode;
+        size++;
+        return true;
+    }
+
+    public boolean addLast(V value) {
+        Node<V> newNode = new Node<>(value);
+        newNode.setPrev(last);
+        if (last != null) {
+            last.setNext(newNode);
+        } else {
+            first = newNode;
+        }
+        last = newNode;
+        size++;
+        return true;
+    }
+
+    public boolean add(int index, V value) {
+        if (index < 0 || index > size) return false;
+        if (index == 0) return addFirst(value);
+        if (index == size) return addLast(value);
+
+        Node<V> current = getNode(index);
+        if (current == null) return false;
+
+        Node<V> newNode = new Node<>(value);
+        Node<V> prev = current.getPrev();
+
+        newNode.setPrev(prev);
+        newNode.setNext(current);
+        if (prev != null) prev.setNext(newNode);
+        current.setPrev(newNode);
+
+        size++;
+        return true;
+    }
+
+    public V getFirst() {
+        return first != null ? first.getValue() : null;
```

Comment:
Returning null can lead to NullPointerException later in the code and hides potential errors